### PR TITLE
[DPC-3916] Split Unit Tests from Integration Tests

### PR DESCRIPTION
**Pull request by chris-ronning-ny**
## 🎫 Ticket

https://jira.cms.gov/browse/DPC-3916

## 🛠 Changes

(What was added, updated, or removed in this PR.)

## ℹ️ Context for reviewers

(Background context, more in-depth details of the implementation, and anything else you'd like to call out or ask reviewers.)

## ✅ Acceptance Validation

(How were the changes verified? Did you fully test the acceptance criteria in the ticket? Provide reproducible testing instructions and screenshots if applicable.)

## 🔒 Security Implications

- [ ] This PR adds a new software dependency or dependencies.
- [ ] This PR modifies or invalidates one or more of our security controls.
- [ ] This PR stores or transmits data that was not stored or transmitted before.
- [ ] This PR requires additional review of its security implications for other reasons.

If any security implications apply, add Jason Ashbaugh (GitHub username: StewGoin) as a reviewer and do not merge this PR without his approval.

### CODE REVIEW

File: lambda/opt-out-import/README.md [hunk 1]
```diff
@@ -19,9 +19,13 @@ configure aws with fake data, `aws configure`
 - run `aws --endpoint-url=http://localhost:4566 s3api list-objects --bucket demo-bucket` to verify that a response file was created
 - run `aws --endpoint-url=http://localhost:4566 s3api get-object --bucket demo-bucket --key T#EFT.ON.ACO.NGD1800.DPRF.D181120.T1000009_response response_file.txt` to download the response file and verify it's correct
 
-## How to run an integration test locally and get a coverage report
+## How to test the build locally and get a coverage report
 - Pull down dpc-app and run `make start-app` or just `docker-compose up start_core_dependencies` to start a local copy of the dpc-consent database.
 - run `docker-compose up -d` in this app's directory
 - run populate_s3.sh
-- run `make integration-test`
+- run `make build-test`
```
Comment (kyeah) [inline lambda/opt-out-import/README.md:26]: This is `test-build` in the makefile 😄

File: lambda/opt-out-import/main_test.go [hunk 2]
```diff
@@ -201,7 +201,7 @@ func TestUploadConfirmationFile(t *testing.T) {
 
 }
 
-func TestDeleteS3File(t *testing.T) {
+func TestIntegrationDeleteS3File(t *testing.T) {
```
Comment (kyeah) [inline lambda/opt-out-import/main_test.go:204]: Also want the `testing.Short()` conditional here to skip if running just unit tests.

Comment (kyeah) [reply lambda/opt-out-import/main_test.go:204]: Although it seems to pass regardless...? 🤔

Comment (kyeah) [reply lambda/opt-out-import/main_test.go:204]: Ah, that actually indicates an issue with the error handling in our code:

```
{"level":"warning","msg":"Unable to delete object: NoCredentialProviders: no valid providers in chain. Deprecated.\n\tFor verbose messaging see aws.Config.CredentialsChainVerboseErrors","time":"2024-03-06T16:31:13.124758226Z"}
```

This error doesn't cause any issues, it just quietly doesn't do anything. If it's a quick change, could we try to adjust that?

**Comment by chris-ronning-ny** [comment]
Good catch on the missed error check @kyeah ! I've made the changes you requested, please review once more!

**Review by kyeah — APPROVED** [review]
ty!

**Pull request merged**